lib/postgres.ts

```diff
@@ -0,0 +1,76 @@
+/**
+ * Library made to simplify accessing / connecting to postgres databases,
+ * and to cleanly handle when the pg module isn't installed.
+ * @author mia-pi-git
+ */
+
+// @ts-ignore in case module doesn't exist
+import type * as PG from 'pg';
+import type {SQLStatement} from 'sql-template-strings';
+import * as Streams from './streams';
+
+export class PostgresDatabase {
+	private pool: PG.Pool;
+	constructor(config = PostgresDatabase.getConfig()) {
+		this.pool = config ? new (require('pg').Pool)(config) : null!;
+	}
+	async query(statement: string | SQLStatement, values?: any[]) {
+		if (!this.pool) {
```

Comment:
Is there any reason to have `pg` be an optional dependency when it is a non-native module?